Add parser for 'show ipv6 interface brief | include {ip}'

palmersample opened the pull request:
## Description
Added a parser for 'show ipv6 interface brief | include {ip}' to emulate IPv4 behavior.

## Motivation and Context
No current parser exists for this functionality. When connecting to a device with Unicon, the management interface check attempts "show ip interface brief | include {ip}" which will not return anything for an IPv6 management interface. This parser emulates the IPv4 functionality to return a dictionary containing the interface(s) matching a given IP address. In the case of a Link-Local address statically assigned to each interface (e.g. fe80::1), the returned dictionary will include all interfaces with the assigned IPv6 Link-Local address.

## Impact (If any)
No impact anticipated or observed

## Screenshots:
<img width="1116" alt="parser_test_result_1" src="https://user-images.githubusercontent.com/40956652/234922490-10e45237-4e53-4dfb-9e51-97ed9313eeba.png">
<img width="1101" alt="parser_test_result_2" src="https://user-images.githubusercontent.com/40956652/234922512-62a66a39-98fb-4a11-b11e-5bbc178b9678.png">
<img width="925" alt="parser_test_result_3" src="https://user-images.githubusercontent.com/40956652/234922528-32913b58-d630-4e06-bb53-e79daf7c91c1.png">


## Checklist:
<!--- This is meant more as a personal checklist so we don't forgot important steps! -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [x] I have updated the changelog.
- [x] I have updated the documentation (If applicable).
- [x] I have added tests to cover my changes (If applicable).
- [x] All new and existing tests passed.
- [x] All new code passed compilation.

dwapstra commented on src/genie/libs/parser/iosxe/show_interface.py:
```diff
@@ -1571,6 +1571,87 @@ def cli(self, ip,output=None):
+    # (nested dict) that has the same data structure across all supported
+    # parsing mechanisms (cli(), yang(), xml()).
+
+    cli_command = 'show ipv6 interface brief'
```
The cli command does not match the documented command with the filter. The command should match the documentation otherwise its confusing for users.

palmersample replied:
Updated the cli_command and verified tests pass as expected

palmersample replied:
Update - although tests pass, expected implementation does not work on live devices (as should have been expected) as the ```| include``` will only return a v6 address and not the interface information. Will work with you on a proposed solution or to determine if there is a more appropriate method to accomplish the intended use case for this parser.

dwapstra commented on src/genie/libs/parser/iosxe/show_interface.py:
```diff
@@ -1571,6 +1571,87 @@ def cli(self, ip,output=None):
+            out = output
+        interface_dict = {}
+
+        # GigabitEthernet1       [up/up]
+        #     FE80::5054:FF:FE18:9003
+        #     2001:DB8:C15:C0::10
```
Please align the comments with the regex compile statements, e.g. line 1 for p1, line 2-3 for p2

palmersample replied:
Updated comment locations to represent the corresponding regular expression

palmersample commented:
Updated commit per conversation with reviewer:

Modified proposed parser and scheme for "show ipv6 interface brief"

Changed parser for "show ipv6 interface brief | include {ip}" to execute a Dq() against the output of "show ipv6 interface brief" parser.

Added test coverage for both classes (test images below)

Tested parsers against a Cisco Catalyst Edge 8000 Virtual with v6-enabled interfaces (static link-local as well as public)

- Result of ```python folder_parsing_job.py -o iosxe -c ShowIpv6InterfaceBrief``` ("show ipv6 interface brief" parser)
<img width="928" alt="sh_ipv6_int_brief_testing" src="https://user-images.githubusercontent.com/40956652/235266945-8600beaf-f02d-4f35-b063-7f4f6f06e3a7.png">

- Result of executing ```pyats parse "show ipv6 interface brief" --testbed-file testbed.yml ``` via CLI:
<img width="1114" alt="CLI-show-ipv6-int-br" src="https://user-images.githubusercontent.com/40956652/235267292-fc7bb94f-2620-4f41-96fd-856e9f30916b.png">

- Result of ```python folder_parsing_job.py -o iosxe -c ShowIpv6InterfaceBriefPipeIp``` ("show ipv6 interface brief | include {ip}" parser)
<img width="929" alt="sh_ipv6_int_brief_pipe_testing" src="https://user-images.githubusercontent.com/40956652/235266975-d7511e0b-9890-4003-8908-96f0b2d0b0d1.png">

- Result of executing ```pyats parse "show ipv6 interface brief | include fe80::1" --testbed-file testbed.yml``` via CLI:
<img width="1117" alt="sh_ipv6_int_br_pipe_ll" src="https://user-images.githubusercontent.com/40956652/235315030-4c4a531f-e606-4845-b536-99cda021328a.png">

- Result of executing ```pyats parse "show ipv6 interface brief | include 2001:db8:dead:beef::2" --testbed-file testbed.yml``` via CLI:
<img width="1115" alt="sh_ipv6_int_br_pipe_gu" src="https://user-images.githubusercontent.com/40956652/235315089-6bc0cb30-4361-4a54-b03a-74185b98b53b.png">
